### ISSUE TITLE
Fix false negatives for disallowInList: true no-duplicate-selectors

### ISSUE THREAD

**Issue by EvgenyOrekhov** [stylelint/stylelint#4339]
> Clearly describe the bug

There are false negatives in `no-duplicate-selectors` with `disallowInList: true`.

> Which rule, if any, is the bug related to?

`no-duplicate-selectors`

> What CSS is needed to reproduce the bug?

```css
textarea {
  border: 1px;
}

input, textarea {
  border: 2px;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-duplicate-selectors": [true, {"disallowInList": true}]
  }
}
```

> Which version of stylelint are you using?

`11.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint 'src/**/*.css'`.

Also this bug reproduces here: https://stylelint.io/demo#N4Igxg9gJgpiBcIAuMAeSCGAnGGAEwAOgHZ54BGEWsW8eAjAA6oDcJAviSQJbGMCuSADR4U6bLgIkylajFp4ATMzbFOxEEJABnAJ7FMqBOG3bN4CMQBm3AObGipPIRBZ+AGxjaXdR2TIuxBAAtFD8jO7cYBgowdownmBIVN4IeADaSG4wIkQgUNzaGO7uEADuAJLEADKFSD6i2ewAutJ46uwg7EA.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

The following warning should be flagged:

```text
5:8 error Unexpected duplicate selector "textarea", first used at line 1 (no-duplicate-selectors)
```

> What actually happened (e.g. what warnings or errors did you get)?

No warnings were flagged.

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

Strangely when these two rules in different order, then warning is shown: https://stylelint.io/demo#N4Igxg9gJgpiBcICWA7ADgVwC4BoAEWMAHlgIYBOMpewAOinngEYTmznx4BMaRA3PQC+9QiQpUa9RizYwOeAIy8BKYShA4QAZwCeKMkQTgtWjeAgoAZkgDmRugzy0Q5DABsYW55weNGzlAgAWigMNDckMFJCIK0YDzAsVi8EPABtLFcYfDoQKCQtUjc3CAB3AEkUABkCrG8CLMEAXSk8NUEQQSA

**Comment by EvgenyOrekhov** [comment]
Appears to be fixed since the demo example is not reproducible anymore.
